rust/main/agents/relayer/src/msg/metadata/base_builder.rs

```diff
@@ -287,4 +273,41 @@ impl BaseMetadataBuilder {
         fetch_storage_locations_helper(validators, &self.cache, &*self.origin_validator_announce)
             .await
     }
+
+    async fn build_and_validate(
+        &self,
+        config: &CheckpointSyncerConf,
+        validator: &H256,
+        latest_index_gauge: Option<IntGauge>,
```

Comment:
It looks like this parameter is always `None`. I wonder why we need it?

Reply:
kept it just in case this would ever need to get called with a different value for `latest_index_gauge`. But I'll remove for now to keep things simple